lib/tests/streamlit/web/server/component_request_handler_test.py

```diff
@@ -67,6 +67,38 @@ def test_success_request(self):
         self.assertEqual(200, response.code)
         self.assertEqual(b"Test Content", response.body)
 
+    def test_outside_component_root_request(self):
+        """Tests to ensure anything outside of the component root is disallowed."""
+
+        with mock.patch("streamlit.components.v1.components.os.path.isdir"):
+            # We don't need the return value in this case.
+            declare_component("test", path=PATH)
+
+        response = self._request_component(
+            "web.server.component_request_handler_test.test//etc/hosts"
+        )
+
+        self.assertEqual(404, response.code)
+        self.assertEqual(b"not found", response.body)
```

Comment:
Can we also add a test here for a parent directory - like, test that `response = self._request_component(
            "web.server.component_request_handler_test.test/../somefile"
        )` returns 404 even if `somefile` is in the parent directory?

Reply:
Heh, jinx @LukasMasuch